onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -155,23 +155,23 @@ static Status GetExternalDataInfo(const ONNX_NAMESPACE::TensorProto& tensor_prot
 // This function does not unpack string_data of an initializer tensor
 static Status ReadExternalDataForTensor(const ONNX_NAMESPACE::TensorProto& tensor_proto,
                                         const ORTCHAR_T* tensor_proto_dir,
-                                        std::unique_ptr<unsigned char[]>& unpacked_tensor,
-                                        SafeInt<size_t>& tensor_byte_size) {
+                                        std::vector<uint8_t>& unpacked_tensor) {
   std::basic_string<ORTCHAR_T> external_file_path;
   onnxruntime::FileOffsetType file_offset;
+  SafeInt<size_t> tensor_byte_size;
   ORT_RETURN_IF_ERROR(GetExternalDataInfo(
       tensor_proto,
       tensor_proto_dir,
       external_file_path,
       file_offset,
       tensor_byte_size));
 
-  unpacked_tensor.reset(new unsigned char[*&tensor_byte_size]);
+  unpacked_tensor.resize(tensor_byte_size);
   ORT_RETURN_IF_ERROR(onnxruntime::Env::Default().ReadFileIntoBuffer(
```

Comment:
>ORT_RETURN_IF_ERROR(onnxruntime::Env::Default().ReadFileIntoBuffer( [](http://example.com/codeflow?start=2&length=67)

Should we check for zero len?

Reply:
Seems zero len is fine here, same in posix version
https://github.com/microsoft/onnxruntime/blob/224380448d371937a084b3d162be7964a702fa9a/onnxruntime/core/platform/windows/env.cc#L255-L256